src/corehost/cli/deps_resolver.cpp

```diff
@@ -119,6 +122,197 @@ void deps_resolver_t::get_dir_assemblies(
     }
 }
 
+bool deps_resolver_t::try_roll_forward(const deps_entry_t& entry,
+    const pal::string_t& probe_dir,
+    pal::string_t* candidate)
+{
+    trace::verbose(_X("Attempting a roll forward for [%s/%s/%s] in [%s]"), entry.library_name.c_str(), entry.library_version.c_str(), entry.relative_path.c_str(), probe_dir.c_str());
+
+    const pal::string_t& lib_ver = entry.library_version;
+
+    fx_ver_t cur_ver(-1, -1, -1);
+    if (!fx_ver_t::parse(lib_ver, &cur_ver, false))
+    {
+        trace::verbose(_X("No roll forward as specified version [%s] could not be parsed"), lib_ver.c_str());
+        return false;
+    }
+
+    // Extract glob string of the form: 1.0.* from the version 1.0.0-prerelease-00001.
+    size_t pat_start = lib_ver.find(_X('.'), lib_ver.find(_X('.')) + 1);
+    pal::string_t maj_min_star = lib_ver.substr(0, pat_start + 1) + _X('*');
+
+    pal::string_t path = probe_dir;
+    append_path(&path, entry.library_name.c_str());
+
+    pal::string_t cache_key = path;
+    append_path(&cache_key, maj_min_star.c_str());
+
+    pal::string_t max_str;
+    if (m_roll_forward_cache.count(cache_key))
```

Comment:
Why do we have .* based roll-forward check? We were only planning to roll-forward based upon patch version.